habitat-baselines/habitat_baselines/rl/ppo/trainer_agent.py

```diff
@@ -0,0 +1,333 @@
+    PointNavResNetNet,
+    PointNavResNetPolicy,
+)
+from habitat_baselines.rl.hrl.hierarchical_policy import (  # noqa: F401.
+    HierarchicalPolicy,
+)
+from habitat_baselines.rl.ppo import PPO  # noqa: F401.
+from habitat_baselines.rl.ppo.policy import NetPolicy
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+@baseline_registry.register_agent
+class TrainerAgent:
+    """
+    A `TrainerAgent` consists of:
+    - Policy: How actions are selected from observations.
+    - Data Storage: How data collected from the environment is stored.
+    - Updater: How the Policy is updated.
+    """
+
+    def __init__(
+        self,
+        config: "DictConfig",
+        env_spec: EnvironmentSpec,
+        is_distrib: bool,
+        device,
+        resume_state: Optional[Dict[str, Any]],
+        num_envs: int,
+        percent_done_fn: Callable[[], float],
+        lr_schedule_fn: Optional[Callable[[float], float]] = None,
+    ):
+        """
+        :param percent_done_fn: Function that will return the percent of the
+            way through training.
+        :param lr_schedule_fn: For a learning rate schedule. ONLY used if
+            specified in the config. Takes as input the current progress in
+            training and returns the learning rate multiplier. The default behavior
+            is to use `linear_lr_schedule`.
+        """
+
+        self._env_spec = env_spec
+        self._config = config
+        self._num_envs = num_envs
+        self._device = device
+        self._ppo_cfg = self._config.habitat_baselines.rl.ppo
+        self._is_distributed = is_distrib
+        self._is_static_encoder = (
+            not config.habitat_baselines.rl.ddppo.train_encoder
+        )
+        self._actor_critic = self._create_policy()
+        self._actor_critic.to(self._device)
+        self._policy_action_space = self._actor_critic.get_policy_action_space(
+            env_spec.action_space
+        )
+        self._updater = self._create_updater(self._actor_critic)
+        if resume_state is not None:
+            self._updater.load_state_dict(resume_state["state_dict"])
+            self._updater.optimizer.load_state_dict(
+                resume_state["optim_state"]
+            )
+
+        logger.info(
+            "agent number of parameters: {}".format(
+                sum(param.numel() for param in self._updater.parameters())
+            )
+        )
+        self.nbuffers = 2 if self._ppo_cfg.use_double_buffered_sampler else 1
+        if lr_schedule_fn is not None:
+            lr_schedule_fn = linear_lr_schedule
+        self._percent_done_fn = percent_done_fn
+
+        self._lr_scheduler = LambdaLR(
+            optimizer=self._updater.optimizer,
+            lr_lambda=lambda x: linear_lr_schedule(
+                percent_done=percent_done_fn()
+            ),
+        )
+
+    def post_init(self, create_rollouts_fn: Optional[Callable] = None) -> None:
```

Comment:
I do not like the name `post_init` or the docstring `Called after the constructor.`
It makes it sound like this is an automatic callback, but it is not, it is called manually by the trainer, which increases code entanglement. 
Is there a reason this cannot be called in __init__ ?
If so, can we call this method by what it does "initialize_rollout" and make sure that the error message in case this method is not called before using the trainer_agent the error message is easy to interpret?